### PR TITLE
Improve mixed types

### PR DESCRIPTION
**Pull request by noamzbr**
condition is now on the ratio of the rare data type being in range

### CODE REVIEW

File: deepchecks/checks/integrity/mixed_types.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def _mixed_types(self, dataset: Union[pd.DataFrame, Dataset], feature_importance
         else:
             display = None
 
-        return CheckResult(result_dict, display=display)
+        return CheckResult(result_dict, display=display, )
```
Comment (shir22) [inline deepchecks/checks/integrity/mixed_types.py:102]: ,?

File: deepchecks/checks/integrity/mixed_data_types.py [hunk 2]
```diff
@@ -9,25 +9,25 @@
...
 
 
-__all__ = ['MixedTypes']
+__all__ = ['MixedDataTypes']
 
 
-class MixedTypes(SingleDatasetBaseCheck):
-    """Search for various types of data in (a) column[s], including hidden mixes in strings.
+class MixedDataTypes(SingleDatasetBaseCheck):
+    """Detect a minority of a data type within a column, such as few string samples in a mostly numeric column.
```
Comment (nirhutnik) [inline deepchecks/checks/integrity/mixed_data_types.py:30]: Minority is a bit weird as its not common to use in these things, IMO.
What about an anomaly?

File: deepchecks/checks/integrity/mixed_data_types.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ class MixedTypes(SingleDatasetBaseCheck):
         ignore_columns (Union[Hashable, List[Hashable]]):
             Columns to ignore, if none given checks based on columns
             variable.
-        n_top_columns (int): (optinal - used only if model was specified)
+        n_top_columns (int): (optional - used only if model was specified)
```
Comment (nirhutnik) [inline deepchecks/checks/integrity/mixed_data_types.py:39]: This makes me happy, thank you

File: deepchecks/checks/integrity/mixed_data_types.py [hunk 4]
```diff
@@ -126,22 +126,32 @@ def is_float(x) -> bool:
 
         return {'strings': strs_pct, 'numbers': nums_pct}
 
-    def add_condition_rare_type_ratio_not_less_than(self, max_rare_type_ratio: float = 0.01):
-        """Add condition - Whether the rarer data type (strings or numbers) have ratio higher than given ratio.
+    def add_condition_rare_type_ratio_not_in_range(self, ratio_range: Tuple[float, float] = (0.01, 0.1)):
+        """Add condition - Whether the ratio of rarer data type (strings or numbers) is not in the "danger zone".
+
+        The "danger zone" represents the following logic - if the rarer data type is, for example, 30% of the data,
+        than the column is presumable supposed to contain both numbers and numeric values. If the rarer data type is,
```
Comment (nirhutnik) [inline deepchecks/checks/integrity/mixed_data_types.py:133]: presumably supposed to
or
presumable to be supposed to

File: deepchecks/checks/integrity/mixed_data_types.py [hunk 5]
```diff
@@ -126,22 +126,32 @@ def is_float(x) -> bool:
...
+        for example, less than 1% of the data, than it's presumably a contamination, but a negligible one. In the range
+        between, there is a real chance that the rarer data type may represent a problem to model training and
+        inference.
 
         Args:
-            max_rare_type_ratio (float): Minimal ratio allowed for the rarer type (numbers or strings)
+            ratio_range (Tuple[float, float]): The range between which the ratio of rarer data type in the column is
+                considered a problem.
```
Comment (nirhutnik) [inline deepchecks/checks/integrity/mixed_data_types.py:140]: why not max and min thresholds? Just out of curiosity.

**Review by nirhutnik — APPROVED** [review]

File: deepchecks/checks/integrity/mixed_data_types.py (hunk 5 above)
Comment (noamzbr) [reply deepchecks/checks/integrity/mixed_data_types.py:140]: mm think this is more compact

**Pull request merged**